### PR TITLE
Update external cluster upgrade procedure

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/ocs_upgrade.py [hunk 1]
```diff
@@ -473,7 +476,17 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
 
     # For external cluster , create the secrets if upgraded version is >= 4.8
     if config.DEPLOYMENT["external_mode"] and upgrade_version >= "4.8":
```
Comment (petr-balogh) [inline ocs_ci/ocs/ocs_upgrade.py:478]: current_version == 4.7 && upgrade_version == 4.8

Comment (vavuthu) [reply ocs_ci/ocs/ocs_upgrade.py]: done

**Comment by vavuthu** [comment]
verification job: https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1437/console

```
19:28:23 - MainThread - ocs_ci.ocs.ocs_upgrade - INFO - CSV name before upgrade is: ocs-operator.v4.7.2
.
.
.
19:31:05 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get csv ocs-operator.v4.8.0-455.ci -n openshift-storage -o yaml
19:31:09 - Thread-2 - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage rsh rook-ceph-tools-external-5658fc685f-hqxlj ceph health detail
19:31:11 - MainThread - ocs_ci.ocs.ocp - INFO - Resource ocs-operator.v4.8.0-455.ci is in phase: Succeeded!


19:38:23 - MainThread - ocs_ci.ocs.resources.storage_cluster - INFO - Verifying ceph health
19:38:23 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc wait --for condition=ready pod -l app=rook-ceph-tools -n openshift-storage --timeout=120s
19:38:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get pod -l 'app=rook-ceph-tools' -o jsonpath='{.items[0].metadata.name}'
19:38:32 - Thread-1 - ocs_ci.utility.retry - WARNING - list index out of range, Retrying in 5 seconds...
19:38:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage exec rook-ceph-tools-external-5658fc685f-hqxlj -- ceph health
19:38:42 - Thread-1 - ocs_ci.utility.retry - WARNING - list index out of range, Retrying in 5 seconds...
19:38:44 - MainThread - ocs_ci.utility.utils - INFO - Ceph cluster health is HEALTH_OK.
```

but the jobfailed at teardown which is not relevant to this PR. hence marked as verified

File: ocs_ci/ocs/ocs_upgrade.py [hunk 2]
```diff
@@ -472,8 +476,22 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
     )
 
     # For external cluster , create the secrets if upgraded version is >= 4.8
-    if config.DEPLOYMENT["external_mode"] and upgrade_version >= "4.8":
-        create_external_secret(ocs_version=upgrade_version, apply=True)
+    if (
+        config.DEPLOYMENT["external_mode"]
+        and original_ocs_version == "4.7"
+        and upgrade_version == "4.8"
```
Comment (jilju) [inline ocs_ci/ocs/ocs_upgrade.py:482]: The conditional statement is not matching with the above comment which says upgraded version is >= 4.8

Comment (vavuthu) [reply ocs_ci/ocs/ocs_upgrade.py:482]: we need update new credentions only when we upgrade from OCS 4.7 to OCS 4.8

Comment (vavuthu) [reply ocs_ci/ocs/ocs_upgrade.py:482]: changed the comment

**Review by jilju — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**